quickwit-metastore/src/metastore/single_file_metastore.rs

```diff
@@ -239,31 +239,44 @@ impl Metastore for SingleFileMetastore {
         Ok(())
     }
 
-    async fn publish_split(&self, index_id: &str, split_id: &str) -> MetastoreResult<()> {
+    async fn publish_splits<'a>(
+        &self,
+        index_id: &str,
+        split_ids: Vec<&'a str>,
+    ) -> MetastoreResult<()> {
         let mut metadata_set = self.get_index(index_id).await?;
 
-        // Check for the existence of split.
-        let split_metadata = metadata_set.splits.get_mut(split_id).ok_or_else(|| {
-            MetastoreErrorKind::SplitDoesNotExist
-                .with_error(anyhow::anyhow!("Split does not exist: {:?}", split_id))
-        })?;
+        let mut updatable_split_ids = vec![];
+        for split_id in split_ids {
+            // Check for the existence of split.
+            let split_metadata = metadata_set.splits.get(split_id).ok_or_else(|| {
```

Comment:
I love that you that you made sure that not mutate the split_metadatas if we are to emit an error.

That being said, the metadata_set you are editing right now is a copy of the actual metadata_set.
We "put" it at the end of the function, so you could have mutated it within the match case if you wanted.

Reply:
I knew it was a copy from `get_index` but I was being paranoid here.